src/code/z_demo.c

```diff
@@ -1,99 +1,1649 @@
+                func_801A246C(4, 0);
+                break;
+
+            case 6:
+                // func_801A246C(SEQ_PLAYER_NATURE, TYPE_2);
+                func_801A246C(4, 2);
+                break;
+
+            case 7:
+                // Audio_GetActiveSequence
+                activeSequence = func_801A8A50(0);
+                break;
+
+            case 8:
+                if (activeSequence != 0xFFFF) {
+                    // Audio_PlayBgmForDayScene
+                    func_801A25E4(activeSequence, dayMinusOne);
+                }
+                break;
+        }
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_demo/func_800EADB0.s")
+#endif
+
+// Command 0x12F: Fade Ambience music
+void Cutscene_Command_FadeAmbienceSequence(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdBase* cmd) {
+    if (csCtx->frames == cmd->startFrame && csCtx->frames < cmd->endFrame) {
+        u8 frameCount = cmd->endFrame - cmd->startFrame;
+
+        Audio_QueueSeqCmd((frameCount << 0x10) | 0x140000FF);
+    }
+}
+
+// Command 0x190: Rumble
+void Cutscene_Command_Rumble(GlobalContext* globalCtx, CutsceneContext* csCtx, CsRumble* cmd) {
+    switch (cmd->type) {
+        case 1:
+            if (csCtx->frames == cmd->startFrame) {
+                func_8013ECE0(0.0f, cmd->unk6, cmd->unk7, cmd->unk8);
+            }
+            break;
+
+        case 2:
+            if ((csCtx->frames >= cmd->startFrame) && (cmd->endFrame >= csCtx->frames) &&
+                ((csCtx->frames == cmd->startFrame) || (globalCtx->state.frames % 64) == 0)) {
+                func_8013ECE0(0.0f, cmd->unk6, cmd->unk7, cmd->unk8);
+            }
+            break;
+    }
+}
+
+// Command 0x9B:
+void Cutscene_Command_FadeColorScreen(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdFadeScreen* cmd) {
+    if ((csCtx->frames >= cmd->startFrame) && (cmd->endFrame >= csCtx->frames)) {
+        f32 alpha;
+
+        globalCtx->envCtx.fillScreen = true;
+        alpha = Environment_LerpWeight(cmd->endFrame, cmd->startFrame, csCtx->frames);
+
+        if (((cmd->unk0 == 1)) || (cmd->unk0 == 2)) {
+            globalCtx->envCtx.screenFillColor[0] = cmd->r;
+            globalCtx->envCtx.screenFillColor[1] = cmd->g;
+            globalCtx->envCtx.screenFillColor[2] = cmd->b;
+
+            if (cmd->unk0 == 2) {
+                globalCtx->envCtx.screenFillColor[3] = (1.0f - alpha) * 255.0f;
+            } else {
+                globalCtx->envCtx.screenFillColor[3] = 255.0f * alpha;
+            }
+        }
+    }
+}
+
+// Command 0x9D: Set Time of Day & Environment Time
+void Cutscene_Command_SetTime(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdDayTime* cmd) {
+    u16 nextTime;
+    u16 hourAsMinutes;
+    u16 minutes;
+
+    if (csCtx->frames == cmd->startFrame) {
+        hourAsMinutes = (cmd->hour * 60.0f) / (60.0f * 24 / 0x10000);
+        minutes = ((cmd->minute + 1) / (60.0f * 24 / 0x10000));
```

Comment:
(60.0f * 24 / 0x10000) might be worthy of a `#define` or macro. It is used in `Sub_S` (thank you by the way, I've been trying to convert the float literal to a decent expression, but couldn't find one), and I think might be useful in `EnBji01` as well. Seems to be what they used when time was in floats.

Reply:
Yeah, sure, I like the idea of this being a macro. Do you have any proposal for the name of the macro?

Reply:
I'm thinking of something like
```c
#define TIME_TO_FLOAT(time) (time * (60.0f * 24 / 0x10000))
#define FLOAT_TO_TIME(time) (time * (0x10000 / 60.0f * 24))
```
Though not sure on the names. For reference 
`Enbji01` would become `tempTimeBeforeMoonCrash < FLOAT_TO_TIME(60)`
and my `SubS` expresion would be `time =
        (((s16)TIME_TO_FLOAT(gSaveContext.time)) % 60) + ((s16)TIME_TO_FLOAT(gSaveContext.time) / 60) * 60)) / 30;`